spec-tests/src/test/java/org/candlepin/spec/PoolSpecTest.java

```diff
@@ -0,0 +1,673 @@
+    //     // implementated at the time of writing.
+
+    //     ApiClient adminClient = ApiClients.admin();
+    //     OwnerDTO owner = this.createOwner(adminClient);
+
+    //     CdnDTO cdn = new CdnDTO()
+    //         .name(this.randomSuffix("test_cdn"))
+    //         .label("Test CDN")
+    //         .url("https://cdn.test.com");
+
+    //     cdn = adminClient.cdns().createCdn(cdn);
+
+    //     String productId = String.valueOf((int) Math.random() * 10000);
+    //     ProductDTO product = new ProductDTO()
+    //         .id(productId)
+    //         .name("test_product-" + productId)
+    //         .addAttributesItem(this.buildAttribute("virt_limit", "10"));
+
+    //     product = adminClient.ownerProducts()
+    //         .createProductByOwner(owner.getKey(), product);
+
+    //     OffsetDateTime startDate = OffsetDateTime.now();
+    //     OffsetDateTime endDate = startDate.plusYears(1);
+
+    //     PoolDTO pool = new PoolDTO()
+    //         .productId(product.getId())
+    //         .quantity(1L)
+    //         .startDate(startDate)
+    //         .endDate(endDate)
+    //         .setCdn(cdn)
+    //         .subscriptionId(this.randomSuffix("source_sub"))
+    //         .subscriptionSubKey("master")
+    //         .upstreamPoolId(this.randomSuffix("upstream_pool_id"));
+
+    //     pool = adminClient.owners()
+    //         .createPool(owner.getKey(), pool);
+
+
+    //     // We should have two pools created, but only the base/primary pool should have the CDN
+    //     // associated with it
+    //     List<PoolDTO> pools = adminClient.pools()
+    //         .listPools(owner.getId(), null, null, true, null);
+
+    //     assertNotNull(pools);
+    //     assertEquals(2, pools.size());
+
+    //     Optional<PoolDTO> bpfilter = pools.stream()
+    //         .filter(p -> "NORMAL".equals(p.getType()))
+    //         .findFirst();
+
+    //     Optional<PoolDTO> dpfilter = pools.stream()
+    //         .filter(p -> !"NORMAL".equals(p.getType()))
+    //         .findFirst();
+
+    //     assertTrue(bpfilter.isPresent());
+    //     assertTrue(dpfilter.isPresent());
+
+    //     PoolDTO basePool = bpfilter.get();
+    //     PoolDTO derivedPool = dpfilter.get();
+
+    //     CdnDTO fetched1 = adminClient.pools().getPoolCdn(basePool.getId());
+    //     assertNotNull(fetched1);
+    //     assertEquals(cdn.getName(), fetched1.getName());
+
+    //     CdnDTO fetched2 = adminClient.pools().getPoolCdn(derivedPool.getId());
+    //     assertNull(fetched2);
+    // }
+
+    @Test
+    public void shouldIncludeProductBranding() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+
+        // Impl note: it is not strictly necessary to have the product ID match anything for the
+        // purposes of this test
+        BrandingDTO brand1 = new BrandingDTO()
```

Comment:
Can you add a factory for random brandings?

Reply:
Done, though I'm not sure how much value it provides.